diff --git a/lib/stm32/common/i2c_common_v2.c b/lib/stm32/common/i2c_common_v2.c
index efdc94ff..c2311592 100644
--- a/lib/stm32/common/i2c_common_v2.c
+++ b/lib/stm32/common/i2c_common_v2.c
@@ -475,6 +475,7 @@ bool i2c_transfer7(uint32_t i2c, uint8_t addr, const uint8_t *w, size_t wn, uint
 
 				} else if (i2c_nack(i2c)) {
 					i2c_clear_nak(i2c);
+					i2c_clear_stop(i2c);
 					status = false;
 					rn = 0;
 					wait = false;
