diff --git a/heatshrink_encoder.c b/heatshrink_encoder.c
index edf4abe..602e177 100644
--- a/heatshrink_encoder.c
+++ b/heatshrink_encoder.c
@@ -235,6 +235,7 @@ HSE_poll_res heatshrink_encoder_poll(heatshrink_encoder *hse,
             break;
         case HSES_FLUSH_BITS:
             hse->state = st_flush_bit_buffer(hse, &oi);
+            return HSER_POLL_EMPTY;
         case HSES_DONE:
             return HSER_POLL_EMPTY;
         default:
