diff --git a/src/cborinternal_p.h b/src/cborinternal_p.h
index ee9c117..02b8294 100644
--- a/src/cborinternal_p.h
+++ b/src/cborinternal_p.h
@@ -52,7 +52,7 @@
  * GCC defines these macros but doesn't support arithmetic including
  * conversions on x86 without SSE2.
  */
-#  if (FLT16_MANT_DIG > 0 || __FLT16_MANT_DIG__ > 0) && \
+#  if ((defined(FLT16_MANT_DIG) && FLT16_MANT_DIG > 0) || (defined(__FLT16_MANT_DIG__) && __FLT16_MANT_DIG__ > 0)) && \
       !(defined(__i386__) && !defined(__SSE2__))
 static inline unsigned short encode_half(float x)
 {
